NitroxModel/Discovery/InstallationFinders/SteamFinder.cs

```diff
@@ -0,0 +1,163 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+using NitroxModel.Discovery.InstallationFinders.Core;
+using NitroxModel.Discovery.Models;
+using static NitroxModel.Discovery.InstallationFinders.Core.GameFinderResult;
+
+namespace NitroxModel.Discovery.InstallationFinders;
+
+/// <summary>
+/// Trying to find the path in the Steam installation directory by the appid that contains the game installation directory.
+/// By default each game will have a corresponding appmanifest_{appid}.acf file in the steamapps folder.
+/// Except for some games that are installed on a different diskdrive, in those case 'libraryfolders.vdf' will give us the real location of the appid folder.
+/// </summary>
+public sealed class SteamFinder : IGameFinder
+{
+    public GameFinderResult FindGame(GameInfo gameInfo)
+    {
+        string steamPath = GetSteamPath();
+
+        if (string.IsNullOrEmpty(steamPath))
+        {
+            return Error("Steam isn't installed");
+        }
+
+        string appsPath = Path.Combine(steamPath, "steamapps");
+        if (File.Exists(Path.Combine(appsPath, $"appmanifest_{gameInfo.SteamAppId}.acf")))
+        {
+            return Ok(new GameInstallation
+            {
+                Path = Path.Combine(appsPath, "common", gameInfo.Name),
+                GameInfo = gameInfo,
+                Origin = GameLibraries.STEAM
+            });
+        }
+
+        string path = SearchAllInstallations(Path.Combine(appsPath, "libraryfolders.vdf"), gameInfo.SteamAppId, gameInfo.Name);
+        if (string.IsNullOrWhiteSpace(path))
+        {
+            return NotFound();
+        }
+
+        return Ok(new GameInstallation
+        {
+            Path = path,
+            GameInfo = gameInfo,
+            Origin = GameLibraries.STEAM
+        });
+    }
+
+    private static string GetSteamPath()
+    {
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            string steamPath = Platforms.OS.Windows.Internal.RegistryEx.Read<string>(@"Software\\Valve\\Steam\SteamPath");
```

Comment:
The mixture of double back quotes and one single seems wrong, is it legit?

Reply:
Good catch. It still worked but no reason to keep it.